sdk/eventhub/event-hubs/test/client.spec.ts

```diff
@@ -10,304 +10,374 @@ import chaiString from "chai-string";
+    properties!["user-agent"].should.endWith(customua);
+    should.equal(properties!.product, "MSJSClient");
+    should.equal(properties!.version, packageVersion);
+    should.equal(properties!.framework, `Node/${process.version}`);
+    should.equal(properties!.platform, `(${os.arch()}-${os.type()}-${os.release()})`);
+    done();
+  });
+});
+
+describe("Errors after close()", function(): void {
+  let client: EventHubClient;
+  let sender: EventHubProducer;
+  let receiver: EventHubConsumer;
+
+  afterEach(() => {
+    return client.close();
   });
 
-  describe("on invalid partition ids like", function(): void {
-    const invalidIds = ["XYZ", "-1", "1000", "-", " "];
-    invalidIds.forEach(function(id: string): void {
-      it(`"${id}" should throw an error`, async function(): Promise<void> {
-        try {
-          client = new EventHubClient(service.connectionString, service.path);
-          await client.getPartitionProperties(id);
-        } catch (err) {
-          debug(`>>>> Received error - `, err);
-          should.exist(err);
-          err.message.should.match(
-            /.*The specified partition is invalid for an EventHub partition sender or receiver.*/gi
-          );
-        }
-      });
+  async function beforeEachTest(entityToClose: string): Promise<void> {
+    should.exist(
+      env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
+      "define EVENTHUB_CONNECTION_STRING in your environment before running integration tests."
+    );
+    should.exist(
+      env[EnvVarKeys.EVENTHUB_NAME],
+      "define EVENTHUB_NAME in your environment before running integration tests."
+    );
+    client = new EventHubClient(env[EnvVarKeys.EVENTHUB_CONNECTION_STRING], env[EnvVarKeys.EVENTHUB_NAME]);
+
+    const timeNow = Date.now();
+
+    // Ensure sender link is opened
+    sender = client.createProducer({ partitionId: "0" });
+    await sender.send({ body: "dummy send to ensure AMQP connection is opened" });
+
+    // Ensure receiver link is opened
+    receiver = client.createConsumer(EventHubClient.defaultConsumerGroup, "0", EventPosition.fromEnqueuedTime(timeNow));
+    const msgs = await receiver.receiveBatch(1, 10);
+    should.equal(msgs.length, 1);
+
+    // close(), so that we can then test the resulting error.
+    switch (entityToClose) {
+      case "client":
+        await client.close();
+        break;
+      case "sender":
+        await sender.close();
+        break;
+      case "receiver":
+        await receiver.close();
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * Tests that each feature of the sender throws expected error
+   */
+  async function testSender(expectedErrorMsg: string): Promise<void> {
+    should.equal(sender.isClosed, true, "Sender is not marked as closed.");
+
+    const testMessage = { body: "test" };
+    let errorSend: string = "";
+    await sender.send(testMessage).catch(err => {
```

Comment:
Just curious, why did you choose to await on the `catch` instead of wrapping the operation in a try/catch?

Reply:
I don't remember!! I used this pattern in Service Bus and copied over the code here